lighthouse-core/test/config/budget-test.js

```diff
@@ -59,136 +60,238 @@ describe('Budget', () => {
     });
 
     it('throws when a duplicate metric type is specified in timings', () => {
-      budget[0].timings.push({metric: 'interactive', budget: 1000});
-      assert.throws(_ => Budget.initializeBudget(budget),
-        /budgets\[0\]\.timings has duplicate entry of type 'interactive'/);
+      input[0].timings.push({metric: 'interactive', budget: 1000});
+      assert.throws(_ => Budget.initializeBudget(input),
+        /has duplicate entry of type 'interactive'/);
+    });
+  });
+
+  describe('path validation', () => {
+    it('with valid inputs', () => {
+      let input = [{path: '/'}];
+      let result = Budget.initializeBudget(input);
+      assert.equal(input[0].path, result[0].path);
+
+      input = [{path: '/*'}];
+      result = Budget.initializeBudget(input);
+      assert.equal(input[0].path, result[0].path);
+
+      input = [{path: '/end$'}];
+      result = Budget.initializeBudget(input);
+      assert.equal(input[0].path, result[0].path);
+
+      input = [{path: '/fish*.php'}];
+      result = Budget.initializeBudget(input);
+      assert.equal(input[0].path, result[0].path);
+
+      input = [{path: '/*.php$'}];
+      result = Budget.initializeBudget(input);
+      assert.equal(input[0].path, result[0].path);
+    });
+
+    it('with invalid inputs', () => {
+      // Missing leading "/"
+      let input = [{path: ''}];
+      assert.throws(_ => Budget.initializeBudget(input), /[Invalid path]/);
+
+      // Missing leading "/"
+      input = [{path: 'cat'}];
+      assert.throws(_ => Budget.initializeBudget(input), /[Invalid path]/);
+
+      // Multiple * characters
+      input = [{path: '/cat*cat*cat'}];
+      assert.throws(_ => Budget.initializeBudget(input), /[Invalid path]/);
+
+      // Multiple $ characters
+      input = [{path: '/cat$cat$'}];
+      assert.throws(_ => Budget.initializeBudget(input), /[Invalid path]/);
+
+      // Invalid $ location
+      input = [{path: '/cat$html'}];
+      assert.throws(_ => Budget.initializeBudget(input), /[Invalid path]/);
+    });
+
+    it('with no path', () => {
+      const input = [{}];
+      const result = Budget.initializeBudget(input);
+      assert.equal(result[0].path, '/');
+    });
+  });
+
+  describe('path matching', () => {
+    it('matches root', () => {
+      assert.ok(Budget.urlMatchesPattern('https://google.com', '/'));
+    });
+
+    it('ignores origin', () => {
+      assert.equal(Budget.urlMatchesPattern('https://go.com/dogs', '/go'), false);
+      assert.equal(Budget.urlMatchesPattern('https://yt.com/videos?id=', '/videos'), true);
+    });
+
+    it('is correct', () => {
+      const pathMatch = (path, pattern) => {
+        const origin = 'https://example.com';
+        return Budget.urlMatchesPattern(origin + path, pattern);
+      };
+
+      // No *, No $:
+      assert.equal(pathMatch('/anything', '/'), true);
+      assert.equal(pathMatch('/anything', '/any'), true);
+      assert.equal(pathMatch('/anything', '/anything1'), false);
```

Comment:
exact match too, just for fun? :)

```suggestion
      assert.equal(pathMatch('/anything', '/anything'), true);
      assert.equal(pathMatch('/anything', '/anything1'), false);
```